parse.go

```diff
@@ -398,15 +398,17 @@ func (p *parser) parse(parsing *Config, tokens <-chan NgxToken, ctx blockCtx, co
 		parsed = append(parsed, stmt)
 
 		// add all comments found inside args after stmt is added
-		for _, comment := range commentsInArgs {
-			comment := comment
-			parsed = append(parsed, &Directive{
-				Directive: "#",
-				Line:      stmt.Line,
-				Args:      []string{},
-				File:      fileName,
-				Comment:   &comment,
-			})
+		if p.options.ParseComments {
```

Comment:
It might be more efficient to do this check above to avoid filling `commentsInArgs` when `!ParseComments`. Did you try doing that?

Reply:
That makes sense. I just added the check much earlier instead of here.